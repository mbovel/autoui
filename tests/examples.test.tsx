import React from "react";
import {
	NoSchemaDeriver,
	defaultComponents,
	useImmerStore,
	storeToProps,
	storeToCommands,
	stateFromJson,
	Auto
} from "../src";
import ReactDOM from "react-dom";

const initialState = stateFromJson({
	firstname: "",
	lastname: "",
	age: 25,
	isAdult: true
});
const derive = NoSchemaDeriver(defaultComponents);

function App() {
	const store = useImmerStore(initialState);
	const props = derive(storeToProps(store), store.dispatch);
	const commands = storeToCommands(store.state, store.dispatch);

	return (
		<main>
			<form>
				<Auto {...props} />
			</form>
			<button onClick={commands.undo}>Undo</button>
			<button onClick={commands.redo}>Undo</button>
		</main>
	);
}

export function renderApp(container: HTMLElement) {
	ReactDOM.render(<App />, container);
}
