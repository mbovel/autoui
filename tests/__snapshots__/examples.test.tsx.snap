// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`minimal renders without crashing 1`] = `
<main>
  <form>
    <p>
       
      <input
        className=""
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
    </p>
    <p>
       
      <input
        className=""
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
    </p>
    <p>
       
      <input
        className=""
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="number"
        value={25}
      />
    </p>
    <p>
       
      <input
        checked={true}
        className=""
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
    </p>
  </form>
  <button
    onClick={[Function]}
  >
    Undo
  </button>
  <button
    onClick={[Function]}
  >
    Undo
  </button>
</main>
`;
